_posts/2021-08-23-gsoc-2021-project.md

```diff
@@ -0,0 +1,116 @@
+
+### Fixing calling `.visualize()` with `filename=None` [#7740](https://github.com/dask/dask/pull/7740)
+
+I fixed a minor bug that was caused when users tried to call `dask.visualize()` with `filename=None`. I fixed it by adding an extra condition before it reaches the error line to make sure the format is not `None` by making it equal to the default `png`.
+
+```python
+import dask
+import dask.array as da
+
+array = da.arange(10)
+dask.visualize(array, filename=None)
+```
+
+<img src="/images/gsoc21/7740.png" alt="#7740 Demo" height=414 width=736>
+
+### Add node size scaling to the Graphviz output for the high level graphs [#7869](https://github.com/dask/dask/pull/7869)
+
+I tweaked the node sizes in the High-level graph's task graphs to align with the density of each layer. Layers with more tasks would appear larger than the rest.
+
+```python
+import dask.array as da
+
+array = da.random.random((10000, 10000), chunks=(200, 200))
+result = array + array.T - array.mean(axis=0)
+
+result.dask.visualize()
+```
+
+<img src="/images/gsoc21/7869.png" alt="#7869 Demo" height=414 width=736>
+
+### Change graphviz font family to sans [#7931](https://github.com/dask/dask/pull/7931)
+
+I modified the font of the Graphviz graphs to `Helvetica` (a sans-serif fontface). This is cleaner and more readable than the default `Times-Roman` (a serif fontface).
+
+```python
+import dask.array as da
+
+array = da.ones((10, 10), chunks=(5, 5))
+array = array + 100
+array = array * 100
+
+array.dask.visualize()
+```
+
+<img src="/images/gsoc21/7931.png" alt="#7931 Demo" height=414 width=736>
+
+### Add tooltips to graphviz [#7973](https://github.com/dask/dask/pull/7973)
+
+I added tooltips to the High-level graph's task graphs. The tooltips show the `layer_type`, `number of tasks`, and the information stored in the `collection_annotations` dictionary for each layer.
+
+```python
+import dask.array as da
+
+x = da.ones((10, 10), chunks=(5, 5))
+x = x + 100
+x = x * 100
+
+x.dask.visualize()
+```
+
+<img src="/images/gsoc21/7973.png" alt="#7973 Demo" height=414 width=736>
+
+### Add colors to represent high level layer types [#7974](https://github.com/dask/dask/pull/7974)
+
+<img src="/images/gsoc21/7974.png" alt="#7974 Demo" height=414 width=736>
+
+## HTML Representation of Task Graphs
+
+I also worked on creating/enhancing HTML reprs for existing classes.
+
+### Add `dask.array` SVG to the HTML Repr [#7886](https://github.com/dask/dask/pull/7886)
+
+I added the Dask Array SVG of the chunks to the HTML Repr of the High-level graph by calling the `dask.array.svg.svg()` function.
+
+```python
+import dask.array as da
+
+array = da.ones((10, 20), chunks=(5, 10))
+array = array.T
+
+array.dask
+```
+
+<img src="/images/gsoc21/7886.png" alt="#7886 Demo" height=414 width=736>
+
+### Add HTML Repr for `Security` Class [#5178](https://github.com/dask/dask/pull/5178)
+
+<img src="/images/gsoc21/5178.png" alt="#5178 Demo" height=414 width=736>
```

Comment:
I don't see this file in the images folder

Reply:
I am waiting for the PR to be merged. I will add the image later.